### PR TITLE
This fixes several issues related to authorization.

### PR DESCRIPTION
**Pull request by walkerp07**
1. Empty account name string passed to 'ProximityBeaconImpl' caused 'getToken()' to fail.
2. There may be cases where several 'UserRecoverableAuthExceptions' need to be handled. To handle all cases, move 'AuthorizedServiceTask' check to 'OnResume()'.

### CODE REVIEW

**Comment by walkerp07** [comment]
I signed it!

File: samples/android/BeaconServiceDemoApp/app/src/main/java/com/google/sample/beaconservice/MainActivityFragment.java [hunk 1]
```diff
@@ -245,6 +245,19 @@ private void createScanner() {
...
 
   @Override
+  public void onResume() {
+    super.onResume();
+
+    // There could be multiple instances when we need to handle 'UserRecoverableAuthExceptions'.
+    // (i.e to updated GMS, to re-login, for permission to use the proximity beacon api)
+    // So run this check every time another activity has finished running.
+    String accountName;
+    if ((accountName = getActivity().getSharedPreferences(Constants.PREFS_NAME, 0).getString("accountName", "")) != "") {
```
Comment (mashbridge) [inline samples/android/BeaconServiceDemoApp/app/src/main/java/com/google/sample/beaconservice/MainActivityFragment.java]: i guess we all typo the != string equality java gotcha once a year. i'll fix it up tho.  :-)

**Comment by mashbridge** [comment]
cheers phillip. not sure why the DS_STORE are here -- the git ignore should have, well, ignored them. no matter, i'll nuke them after i merge.

File: samples/android/BeaconServiceDemoApp/app/src/main/java/com/google/sample/beaconservice/MainActivityFragment.java (hunk 1 above)
Comment (walkerp07) [inline samples/android/BeaconServiceDemoApp/app/src/main/java/com/google/sample/beaconservice/MainActivityFragment.java]: Oh, yep.... opps. I've been working with Go a lot lately and must of mixed things up. Thanks :)

**Comment by walkerp07** [comment]
Acknowledged

**Pull request merged**